### PR TITLE
AR features toggle

### PR DESCRIPTION
**Pull request by IodaMikeMapbox**
Checks:

* [ ] Update changelog
* [x] Rebase to `dev` branch
* [x] Assign reviewers

Linked issues:
ios part of mapbox/mapbox-vision#2024

### CODE REVIEW

File: demo/ARContainerViewController.swift [hunk 1]
```diff
@@ -6,19 +6,55 @@ import UIKit
...
+        case .lane:
+            return .fence
+        case .fence:
+            return .laneAndFence
+        case .laneAndFence:
+            return .lane
+        }
+    }
+
+    var isLane: Bool {
```
Comment (chezzdev) [inline demo/ARContainerViewController.swift]: How about `containsLane`?

File: demo/ARContainerViewController.swift [hunk 2]
```diff
@@ -6,19 +6,55 @@ import UIKit
...
+        }
+    }
 
     override func viewDidLoad() {
         super.viewDidLoad()
         mapViewController.completion = present
 
         presentMap()
 
+        let featuresSwitchButton = UIButton(frame: CGRect(x: 0, y: 0, width: arViewController.view.frame.size.width, height: arViewController.view.frame.size.height))
```
Comment (chezzdev) [inline demo/ARContainerViewController.swift]: Let's use Autolayout here.

Comment (chezzdev) [inline demo/ARContainerViewController.swift]:
```suggestion
        let featureSwitchButton = UIButton(frame: CGRect(x: 0, y: 0, width: arViewController.view.frame.size.width, height: arViewController.view.frame.size.height))
```

File: demo/ARContainerViewController.swift [hunk 3]
```diff
@@ -35,6 +71,11 @@ final class ARContainerViewController: UIViewController {
         ])
     }
 
+    @objc
+    func switchARFeature(_ sender: Any) {
```
Comment (chezzdev) [inline demo/ARContainerViewController.swift]: Seems like we don't need a param here.

File: demo/ARRoute+DirectionsRoute.swift [hunk 4]
```diff
@@ -28,3 +28,46 @@ extension MapboxVisionARNative.Route {
                   destinationStreetName: route.legs.last?.destination.name ?? "")
     }
 }
+
+func convert(maneuverType: MapboxDirections.ManeuverType) -> MapboxVisionARNative.ManeuverType {
```
Comment (chezzdev) [inline demo/ARRoute+DirectionsRoute.swift]: WDYT about an extension on `MapboxDirections.ManeuverType` for this?

File: demo/ARContainerViewController.swift (hunk 1 above)
Comment (IodaMikeMapbox) [reply demo/ARContainerViewController.swift]: changed

File: demo/ARContainerViewController.swift (hunk 2 above)
Comment (IodaMikeMapbox) [reply demo/ARContainerViewController.swift]: done

Comment (IodaMikeMapbox) [reply demo/ARContainerViewController.swift]: renamed

File: demo/ARContainerViewController.swift (hunk 3 above)
Comment (IodaMikeMapbox) [reply demo/ARContainerViewController.swift]: removed

File: demo/ARRoute+DirectionsRoute.swift (hunk 4 above)
Comment (IodaMikeMapbox) [reply demo/ARRoute+DirectionsRoute.swift]: changed

File: demo/ARContainerViewController.swift [hunk 5]
```diff
@@ -6,19 +6,62 @@ import UIKit
...
+    }
+}
+
 final class ARContainerViewController: UIViewController {
     private lazy var mapViewController = ARMapNavigationController()
     private lazy var arViewController = VisionARViewController()
 
     weak var navigationDelegate: NavigationDelegate?
     private var navigationService: NavigationService?
+    private var activeARFeature: ARFeature = .lane {
```
Comment (dersim-davaod) [inline demo/ARContainerViewController.swift]: Do we want to have a case to disable both lane and fence (make them invisible)?

File: demo/ARContainerViewController.swift [hunk 6]
```diff
@@ -6,19 +6,62 @@ import UIKit
...
 
     override func viewDidLoad() {
         super.viewDidLoad()
         mapViewController.completion = present
 
         presentMap()
 
+        let featureSwitchButton = UIButton()
+        featureSwitchButton.translatesAutoresizingMaskIntoConstraints = false
+        featureSwitchButton.setTitle("", for: .normal)
```
Comment (dersim-davaod) [inline demo/ARContainerViewController.swift]: Do we want to update title to display current case explicitly?

File: demo/ARContainerViewController.swift (hunk 5 above)
Comment (IodaMikeMapbox) [reply demo/ARContainerViewController.swift]: no, as I got from the issue

File: demo/ARContainerViewController.swift (hunk 6 above)
Comment (IodaMikeMapbox) [reply demo/ARContainerViewController.swift]: no, this button is invisible and сovers all the space of the screen, there is a plan to show the state but in the future versions, no design for now

**Pull request merged**